Sources/MapboxMaps/Style/Style+Localization.swift

```diff
@@ -40,20 +51,33 @@ extension Style {
 
     /// Filters through source to determine supported locale styles.
     /// This is needed for v7 support
-    internal func getLocaleValue(locale: Locale) -> String {
+    internal func getLocaleValue(locale: Locale) -> String? {
+        // Reference for supported languages: https://docs.mapbox.com/ios/navigation/guides/localization-and-internationalization/#supported-languages
+        let supportedLocaleIdentifiers = ["ar", "zh", "en", "fr", "de", "it", "ja", "ko", "pt", "ru", "zh-Hans", "es", "zh-Hant", "vi"]
+
+        // Do nothing if we do not support the locale
+        if !supportedLocaleIdentifiers.contains(locale.languageCode!) {
+            return nil
+        }
+
         let vectorSources = allSourceIdentifiers.filter { source in
             return source.type == .vector
         }
 
         for sourceInfo in vectorSources {
             do {
-                let tempSource = try source(withId: sourceInfo.id) as VectorSource
+                if locale.identifier.starts(with: "zh") {
+                    let tempSource = try source(withId: sourceInfo.id) as VectorSource
```

Comment:
why is this variable named `tempSource`? What's `temp` about it?

Reply:
I've always used 'temp` for throw away vars.